addons/hr_timesheet/views/project_task_views.xml

```diff
@@ -221,12 +221,20 @@
                     <attribute name="js_class">hr_timesheet_graphview</attribute>
                 </xpath>
                 <xpath expr="//field[@name='stage_id']" position='after'>
-                    <field name="allocated_hours" widget="timesheet_uom"/>
-                    <field name="remaining_hours" widget="timesheet_uom"/>
-                    <field name="effective_hours" widget="timesheet_uom"/>
-                    <field name="total_hours_spent" widget="timesheet_uom"/>
-                    <field name="overtime" widget="timesheet_uom"/>
-                    <field name="subtask_effective_hours" widget="timesheet_uom"/>
+                    <field name="allocated_hours" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="allocated_hours" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="remaining_hours" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="remaining_hours" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="effective_hours" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="effective_hours" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="total_hours_spent" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="total_hours_spent" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="overtime" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="overtime" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="subtask_effective_hours" widget="timesheet_uom" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="subtask_effective_hours" widget="timesheet_uom" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="progress" groups="hr_timesheet.group_hr_timesheet_user"/>
+                    <field name="progress" invisible="1" groups="!hr_timesheet.group_hr_timesheet_user"/>
```

Comment:
I understand that you need the fields for the inheritance of views, but I'm not sure duplicating the fields like that is a very good practice. e.a. if a widget is removed/updated, you have to be carefull and check twice as many fields. A more odoo-like way to handle cases like this would be to use a boolean computed field.
something like 
```py
has_timesheet_rigths = fields.boolean(compute='compute_field')
```
Then in your view
```xml
<field name="progress" invisible="has_timesheet_rigths" />
```

Reply:
After discussing with XBO, it's not possible for the "invisible" field to interpret another value than 0 or 1. The reason is that pivot/grid/graph views perform aggregations so that the records are grouped and cannot be processed individually (like it would be the case in tree/form/kanban views). 
So even if the current solution is a bit ugly, I'm afraid there is no alternative.